tests/containers/podman_netavark.pm

```diff
@@ -0,0 +1,174 @@
+    assert_script_run("podman run --network $net2->{name}:ip=$ctr2->{ip},mac=$ctr2->{mac} --network $net1->{name}:ip=$ctr2->{ip_sec},mac=$ctr2->{mac_sec} -dt --name $ctr2->{name} $ctr2->{image}");
+
+    # second container should have 2 interfaces
+    my $net1_reg = qr@ether\s+$ctr2->{mac}.*\s+inet\s+$ctr2->{ip}\/16@;
+    my $net2_reg = qr@ether\s+$ctr2->{mac_sec}.*\s+inet\s+$ctr2->{ip_sec}\/16@;
+    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth0'", sub { /$net1_reg|$net2_reg/m });
+    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth1'", sub { /$net1_reg|$net2_reg/m });
+
+    # busybox container should be able to resolve apache container
+    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'nslookup $ctr1->{name}'", sub { /Name:\s+$ctr1->{name}\s+Address.*$ctr1->{ip}/m });
+    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'wget -S $ctr1->{ip}:80'", sub { /HTTP.* 200 OK/ });
+    assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+    assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+    assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net2->{name}");
+    remove_subtest_setup;
+
+    ## TEST3
+    record_info('TEST3', 'create a dual stack network');
+    $net1->{name} = 'test_dual_stack';
+    assert_script_run("podman network create --ipv6 --gateway $net1->{gateway_v6} --subnet $net1->{subnet_v6} --gateway $net1->{gateway} --subnet $net1->{subnet} $net1->{name}");
+    assert_script_run("podman run --network $net1->{name} -d --name $ctr1->{name6} --ip6 $ctr1->{ip6} -p 8080:80 $ctr1->{image}");
+    assert_script_run("podman run --network $net1->{name} -d --name $ctr1->{name} --ip $ctr1->{ip} -p 8888:80 $ctr1->{image}");
+
+    foreach my $req ((
+            "-6 http://[$ctr1->{ip6}]:80",
+            "-4 http://$ctr1->{ip}:80",
+            'http://localhost:8080',
+            'http://localhost:8888'
+    )) {
+        validate_script_output("curl --retry 5 --retry-all-errors --head --silent $req", sub { /HTTP.* 200 OK/ }, timeout => 120);
+    }
+
+    assert_script_run("podman container inspect $ctr1->{name} --format {{.NetworkSettings.Networks.$net1->{name}.IPAddress}}");
+    assert_script_run("podman container inspect $ctr1->{name6} --format {{.NetworkSettings.Networks.$net1->{name}.IPAddress}}");
+    assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+    assert_script_run("grep $ctr1->{name6} /run/containers/networks/aardvark-dns/$net1->{name}");
+    remove_subtest_setup;
+
+    #only for netavark v1.6+
+    my $cur_version = script_output('rpm -q --qf "%{VERSION}\n" netavark');
+    if (!is_jeos && package_version_cmp($cur_version, '1.6.0') >= 0) {
```

Comment:
why `!is_jeos` ?

Reply:
`jeos`'s kernel does not have `macvlan` kernel module

Reply:
ok, was about to try `modprobe macvlan` but if it's not there.. Thanks!

Reply:
Maybe write there a comment.

Reply:
done